libraries/core_libs/consensus/src/pbft/block_proposer.cpp

```diff
@@ -45,16 +45,43 @@ bool SortitionPropose::propose() {
         return false;
       }
     } else {
-      LOG(log_dg_)
-          << "Will not propose DAG block, will reset number of tries. Get difficulty at stale , current propose level "
-          << propose_level;
+      LOG(log_dg_) << "Will not propose DAG block, will reset number of tries. Get difficulty at stale , current "
+                      "propose level "
+                   << propose_level;
       last_frontier_ = frontier;
       num_tries_ = 0;
       return false;
     }
   }
 
-  vdf.computeVdfSolution(sortition_params, frontier.pivot.asBytes());
+  std::future<void> result = std::async(std::launch::async, &vdf_sortition::VdfSortition::computeVdfSolution, &vdf,
+                                        sortition_params, frontier.pivot.asBytes());
+  bool vdf_computation_cancel = false;
+  while (result.wait_for(std::chrono::milliseconds(100)) != std::future_status::ready) {
+    auto latest_frontier = dag_mgr_->getDagFrontier();
+    if (!latest_frontier.isEqual(frontier)) {
+      if (vdf.isStale(sortition_params)) {
+        vdf_computation_cancel = true;
+        break;
+      } else {
+        auto latest_level = proposer->getProposeLevel(latest_frontier.pivot, latest_frontier.tips) + 1;
+        if (latest_level > propose_level) {
+          vdf_computation_cancel = true;
+          break;
+        }
+      }
+    }
+  }
+
+  if (vdf_computation_cancel) {
+    last_frontier_ = frontier;
+    num_tries_ = 0;
+    vdf.cancelCompute();
+    result.wait();
+    // Since compute was canceled there is a chance to propose a new block immediately, return true to skip sleep
+    return true;
```

Comment:
Please see my comments below, I think here should be return false

Reply:
When we cancel compute, I want to try to propose a new block immediately. So I think it should be true